Rubberduck.SmartIndenter/AbsoluteCodeLine.cs

```diff
@@ -120,58 +120,34 @@ public string Escaped
 
         public string EndOfLineComment { get; private set; }
 
-        public IEnumerable<string> Segments
-        {
-            get { return _segments; }
-        }
+        public IEnumerable<string> Segments => _segments;
 
         public string ContinuationRebuildText
         {
             get
             {
-                var output = (_code + " " + EndOfLineComment).Trim();
+                var output = ($"{_code} {EndOfLineComment}").Trim();
                 return HasContinuation ? output.Substring(0, output.Length - 1) : output;
             }
         }
 
-        public bool ContainsOnlyComment
-        {
-            get { return _code.StartsWith("'") || _code.StartsWith("Rem "); }
-        }
+        public bool ContainsOnlyComment => _code.StartsWith("'") || _code.StartsWith("Rem ");
 
-        public bool IsDeclaration
-        {
-            get { return !IsEmpty && (!IsProcedureStart && !ProcedureStartIgnoreRegex.IsMatch(_code)) && DeclarationRegex.IsMatch(_code); }
-        }
+        public bool IsDeclaration => !IsEmpty && (!IsProcedureStart && !ProcedureStartIgnoreRegex.IsMatch(_code)) && DeclarationRegex.IsMatch(_code);
 
         public bool IsDeclarationContinuation { get; set; }
 
-        public bool HasDeclarationContinuation
-        {
-            get
-            {
-                return (!IsProcedureStart && !ProcedureStartIgnoreRegex.IsMatch(_code)) &&
-                       !ContainsOnlyComment &&
-                       string.IsNullOrEmpty(EndOfLineComment) &&
-                       HasContinuation &&
-                       ((IsDeclarationContinuation && Segments.Count() == 1) || DeclarationRegex.IsMatch(Segments.Last()));
-            }
-        }
+        public bool HasDeclarationContinuation => (!IsProcedureStart && !ProcedureStartIgnoreRegex.IsMatch(_code)) &&
+                                                  !ContainsOnlyComment &&
+                                                  string.IsNullOrEmpty(EndOfLineComment) &&
+                                                  HasContinuation &&
+                                                  ((IsDeclarationContinuation && Segments.Count() == 1) || DeclarationRegex.IsMatch(Segments.Last()));
```

Comment:
This code is too hard to follow for me.

1) I would have the expressions in its own block, like this:
```
public bool HasDeclarationContinuation =>
     <Holy be all end all giant jumbo expression>
```

2) I would have each condition on their own lines. The first line has 2 conditions while subsequent lines has only one conditional. The last line need more handling - either put them in a new indent block to split the `||` condition into their own line.